packages/aws-cdk-lib/aws-s3-deployment/lib/bucket-deployment.ts

```diff
@@ -869,3 +872,10 @@ export interface UserDefinedObjectMetadata {
    */
   readonly [key: string]: string;
 }
+
+function sourceConfigEqual(stack: cdk.Stack, a: SourceConfig, b: SourceConfig) {
```

Comment:
safer to check the equality of stringified `SourceConfig` minus `bucket`? If we add a new propr to `SourceConfig` in the future then this function will silently be inaccurate. Although `SourceConfig` hasn't changed in 4 years... not a big deal

```ts
JSON.stringify({...a, bucket: undefined}) === JSON.stringify({...b, bucket: undefined})
```

Reply:
Ok, this isn't really supposed to be a blocker. I will leave it up as an open question and if you agree you can open a new PR later.